pyclowder/api/v2/datasets.py

```diff
@@ -0,0 +1,386 @@
+
+    result = requests.get(url, headers=headers,
+                          verify=connector.ssl_verify if connector else True)
+    result.raise_for_status()
+
+    return json.loads(result.text)
+
+
+def get_file_list(connector, host, key, datasetid):
+    """Get list of files in a dataset as JSON object.
+
+    Keyword arguments:
+    connector -- connector information, used to get missing parameters and send status updates
+    host -- the clowder host, including http and port, should end with a /
+    key -- the secret key to login to clowder
+    datasetid -- the dataset to get filelist of
+    """
+    headers = {"Authorization": "Bearer " + key}
+
+    url = "%sapi/v2/datasets/%s/files" % (host, datasetid)
+
+    result = requests.get(url, headers=headers, verify=connector.ssl_verify if connector else True)
+    result.raise_for_status()
+
+    return json.loads(result.text)
+
+
+def remove_metadata(connector, host, key, datasetid, extractor=None):
+    """Delete dataset JSON-LD metadata from Clowder.
+
+    Keyword arguments:
+    connector -- connector information, used to get missing parameters and send status updates
+    host -- the clowder host, including http and port, should end with a /
+    key -- the secret key to login to clowder
+    datasetid -- the dataset to fetch metadata of
+    extractor -- extractor name to filter deletion
+                    !!! ALL JSON-LD METADATA WILL BE REMOVED IF NO extractor PROVIDED !!!
+    """
+    headers = {"Authorization": "Bearer " + key}
+
+    filterstring = "" if extractor is None else "&extractor=%s" % extractor
+    url = '%sapi/v2/datasets/%s/metadata' % (host, datasetid)
+
+    # fetch data
+    result = requests.delete(url, stream=True, headers=headers,
+                             verify=connector.ssl_verify if connector else True)
+    result.raise_for_status()
+
+
+def submit_extraction(connector, host, key, datasetid, extractorname):
+    """Submit dataset for extraction by given extractor.
+
+    Keyword arguments:
+    connector -- connector information, used to get missing parameters and send status updates
+    host -- the clowder host, including http and port, should end with a /
+    key -- the secret key to login to clowder
+    datasetid -- the dataset UUID to submit
+    extractorname -- registered name of extractor to trigger
+    """
+    headers = {'Content-Type': 'application/json',
+                "Authorization": "Bearer " + key}
+
+    url = "%sapi/v2/datasets/%s/extractions?key=%s" % (host, datasetid)
+
+    result = requests.post(url,
+                           headers=headers,
+                           data=json.dumps({"extractor": extractorname}),
+                           verify=connector.ssl_verify if connector else True)
+    result.raise_for_status()
+
+    return result.status_code
+
+
+# TODO not implemented in v2 yet
+def submit_extractions_by_collection(connector, host, key, collectionid, extractorname, recursive=True):
```

Comment:
can we just remove endpoints from this file that arent implemented in v2 for clarity? it isn't for certain that we'll ever have Collections etc. in v2

Reply:
I am in favor as well.